### PR TITLE
Adapt for rails 4.1 and do some little cleanup

### PR DESCRIPTION
**Pull request by franciscoj**
#### 🎩 What?

Update Immortal to work with rails 4.1

### CODE REVIEW

File: lib/immortal.rb [hunk 1]
```diff
@@ -57,26 +61,12 @@ def where_with_deleted(conditions)
...
-    def find_with_deleted(*args)
-      ActiveSupport::Deprecation.warn('[immortal] we are deprecating #find_with_deleted use where_with_deleted instead')
-      without_default_scope do
-        find(*args)
-      end
-    end
-
     def where_only_deleted(conditions)
       without_default_scope do
         where(deleted: true).where(conditions)
```
Comment (jhbabon) [inline lib/immortal.rb]: We should use `COLUMN_NAME` here as well.

Comment (jhbabon) [inline lib/immortal.rb]: O even better, we could add a `mortal` scope that does that for you so we don't add the `where` clause all the time.

Comment (franciscoj) [inline lib/immortal.rb]: @jhbabon done!

File: lib/immortal.rb [hunk 2]
```diff
@@ -121,20 +117,33 @@ def destroy!
     end
 
     def destroy_without_callbacks
-      self.class.unscoped.where(id: id).update_all(deleted: true, updated_at: current_time_from_proper_timezone)
+      scoped_record.update_all(
```
Comment (jhbabon) [inline lib/immortal.rb]: You can chain scopes:

```ruby
scoped_record.immortal.update_all(updated_at: current_time_from_proper_timezone)
```

Comment (franciscoj) [inline lib/immortal.rb]: Yup, but that doesn't work. I'm not scoping here but updating the scoped stuff.

File: lib/immortal.rb [hunk 3]
```diff
@@ -1,10 +1,14 @@
 require 'immortal/belongs_to'
 
+# Include +Immortal+ module to activate soft deleted on your model.
```
Comment (saimonmoore) [inline lib/immortal.rb]: `s/deleted/delete`

File: lib/immortal.rb (hunk 2 above)
Comment (jhbabon) [inline lib/immortal.rb]: True. The `#update_all` signature it's confusing sometimes.

**Pull request merged**